src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -115,6 +113,8 @@ private sealed class HttpClientParameters
             public readonly RemoteCertificateValidationCallback? ServerCertificateValidationCallback;
             public readonly X509CertificateCollection? ClientCertificates;
             public readonly CookieContainer? CookieContainer;
+            public readonly ServicePoint? ServicePoint;
```

Comment:
We should likely be excluding parameters with a ServicePoint from caching (update `AreParametersAcceptableForCaching`), or we might limit the global cache to a single host.

On a side note, I'm not a fan of how we're doing client caching here overall, it's trivial to fall off a cliff from 100% to 0% client reuse. If we're looking at making WebRequest less bad, this is something we could look at improving.